### ISSUE TITLE
error on commas in AWS Tag values

### ISSUE THREAD

**Issue by klauern** [terraform-linters/tflint#788]
There's a formatting specification for Tag Keys and Values.  One issue with applying some changes is that you are disallowed from using certain characters (notably the `,`) in a Tag Value, and this is not being caught by `tflint` currently.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-resource-tags.html#w2ab1c25c10d254c13c17

**Comment by RyanDhuse** [comment]
I tripped on this one today with the "#" character.  I'd be happy to put some effort into a merge request.  Do the maintainers have some general direction on how this should be done?

**Comment by bendrucker** [comment]
Hey, working on making tags rules easier to write. The existing tags rule (missing_tags) is in pretty rough shape. https://github.com/terraform-linters/tflint/pull/803 should lay the ground work. It adds a `tags` package within `awsrules` for dealing with tags that other tag-related rules can use.

https://github.com/terraform-linters/tflint/blob/master/rules/awsrules/aws_resource_missing_tags.go

I can see this being a `aws_invalid_tags` rule. In the mean time, might be helpful to start thinking about how to actually do this validation. If aws-sdk-go can do it (or even if not), you might also look into implementing this as field validator in terraform-provider-aws. TFLint should ideally suggest optional best practices and leave static validation to providers. The one exception is when a network connection would be required for validation, which Terraform validation forbids.

**Comment by wata727** [comment]
Yes, I think static validation should ideally be included in providers.
On the other hand, I think there are some challenges in static validations by providers:

- Since it is often performed in a later process such as CI, it is difficult to try easily.
- Validation error cannot be ignored, so it is difficult to incorporate specifications that change frequently.

I think providers should be able to solve this kind of problem in the future, but until then TFLint doesn't seem to be a bad idea to tackle this problem. So I'm positive about implementing this kind of rule.
